group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2451,6 +2451,8 @@ private CoordinatorResult<Void, Record> completeClassicGroupJoin(
 
         if (group.isInState(DEAD)) {
             log.info("Group {} is dead, skipping rebalance stage.", groupId);
+        } else if (!containsClassicGroup(group.groupId())) {
+            log.info("Group {} is null or not a classic group, skipping rebalance stage.", groupId);
```

Comment:
I find this check a little unexpected here because the method received a `ClassicGroup`. It seems that `completeClassicGroupJoin` is called from two different contexts. In the first one, it is called in the regular processing of the request so the `ClassicGroup` is available and we know that it is fine. In the second one, it is called from an expired timer. In this case, we need to validate that the group still exists. Therefore, I wonder whether we should add overload to `completeClassicGroupJoin` which takes the group id and looks up the group based on it. We could then use it in the "timer context". What do you think?

Reply:
Yeah I agree with it. I'll try to change the params and overload them if they are also called in the regular context